lib/deppack.js

```diff
@@ -0,0 +1,289 @@
+'use strict';
+const sysPath = require('path');
+const fs = require('fs');
+const detective = require('detective');
+const browserResolve = require('browser-resolve');
+const requireDefinition = require('commonjs-require-definition');
+const each = require('async-each');
+
+const shims = {
+  'assert': {},
+  'buffer': {},
+  'child_process': {},
+  'cluster': {},
+  'crypto': {},
+  'dgram': {},
+  'dns': {},
+  'events': {},
+  'fs': {},
+  'http': {},
+  'https': {},
+  'net': {},
+  'os': {},
+  'path': {},
+  'process': { env: { NODE_ENV: process.env.NODE_ENV } },
+  'punycode': {},
+  'querystring': {},
+  'readline': {},
+  'repl': {},
+  'string_decoder': {},
+  'tls': {},
+  'tty': {},
+  'url': {},
+  'util': {},
+  'vm': {},
+  'zlib': {}
+};
+
+const id = x => x;
+const uniq = arr => arr.filter((i, pos) => arr.indexOf(i) === pos);
+const not = f => i => !f(i);
+
+const readFile = (path, callback) => {
+  return fs.readFile(path, {encoding: 'utf8'}, callback);
+};
+
+const getModuleRootPath = path => {
+  const split = path.split('/');
+  const index = split.lastIndexOf('node_modules');
+  return split.slice(0, index + 2).join('/');
+};
+
+const getModuleRootName = path => {
+  const split = path.split('/');
+  const index = split.lastIndexOf('node_modules');
+  return split[index + 1];
+};
+
+const getModuleMainPath = path => sysPath.dirname(getMainFile(path));
+const packageJson = filePath => require(relativeToRoot(filePath, 'package.json'));
+const browserMappings = filePath => packageJson(filePath).browser || {};
+const relativeToRoot = (filePath, relPath) => sysPath.join(getModuleRootPath(filePath), relPath);
+
+const isRelative = path => path.slice(0, 2) === './';
+
+const globalBrowserMappings = filePath => {
+  const brMap = browserMappings(filePath);
+
+  const newBrMap = {};
+
+  Object.keys(brMap).filter(not(isRelative)).forEach(key => {
+    const val = brMap[key];
+    newBrMap[key] = generateModuleName(relativeToRoot(filePath, val));
+  });
+
+  return newBrMap;
+};
+
+const expandedFilePath = filePath => {
+  const brMap = browserMappings(filePath);
+
+  Object.keys(brMap).filter(isRelative).forEach(key => {
+    const val = brMap[key];
+    if (filePath === relativeToRoot(filePath, val)) {
+      filePath = relativeToRoot(filePath, key);
```

Comment:
We should probably stop the iteration here. Can we use `for-in & break`? Or, `Array#some`.